portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,30 +19,56 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.List;
 import java.util.ArrayList;
 import com.google.gson.Gson;
 import com.google.sps.data.Comment;
+import com.google.sps.data.CommentRequest;
 import java.util.Date;
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment").addSort("date", SortDirection.DESCENDING);
+    String num = request.getParameter("num");
+    String order = request.getParameter("order");
+    String user = request.getParameter("user").trim().toUpperCase();
+    
+    int displayNum = Integer.MAX_VALUE;
+    if(!num.equals("all")) {
+      displayNum = Integer.parseInt(num);
+    }
+
+    SortDirection sort = SortDirection.DESCENDING;
+    if(order.equals("oldest")) {
+      sort = SortDirection.ASCENDING;
+    }
+
+    Query query = new Query("Comment").addSort("date", sort);
+
+    if(user != null && !user.equals("")) {
+      query = new Query("Comment").addSort("date", sort).setFilter(new FilterPredicate("username", FilterOperator.EQUAL, user));
```

Comment:
nit: this line is long to the point where I need to scroll to read it. Can you try to format your files to set max line length to what I assume is 100 characters.